### PR TITLE
Feature: email notification

### LINKED ISSUE #2: Add basic email utility class for error notifications
## Requirements:

- Make a Notification utility class
- This class has private variables email and password
- Add a basic html template that would be sent to the user.

### PR DESCRIPTION
**Pull request by PhoenixOutlaw**

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,2 +1,4 @@
 TOKEN=YOUR_TOKEN
 DATABASE_URL=mongodb://localhost:27017
+SERVICE_EMAIL = "dealfinder@gmail.com"
+SERVICE_PASSWORD ="16digitgpass"
```
Comment (BajajGirik) [inline .env.example]: Let's make it a little more generic and not add specific email and password in the example.

```
SERVICE_EMAIL=YOUR_SERVICE_EMAIL
SERVICE_EMAIL_PASSWORD=YOUR_SERVICE_EMAIL_PASSWORD
```

File: src/notfication/notify.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+from email.message import EmailMessage
+import smtplib
+import os
+import ssl
+# from dotenv import load_dotenv
+# load_dotenv()
```
Comment (BajajGirik) [inline src/notfication/notify.py]: Should this be commented?

File: src/notfication/notify.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+class Notify:
+    # __serviceAccEmail = os.getenv('SERVICE_EMAIL')
+    # __serviceAccPassword = os.getenv('SERVICE_PASSWORD')
```
Comment (BajajGirik) [inline src/notfication/notify.py]: I see that you're referencing these variables later in the `email` method so why are they commented?

File: src/notfication/notify.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    _subject = " Something went wrong !!"
+
+    def email(errorMessage:str,to:list[str]):
```
Comment (BajajGirik) [inline src/notfication/notify.py]: Let's rename the method to `sendEmail` to better specify it's functionality.

Comment (BajajGirik) [reply src/notfication/notify.py]: I know that we will only be using this class for error notifications but rather than specifying the parameter name as `errorMessage`, let's make it `message`  to be a little more generic

Comment (BajajGirik) [reply src/notfication/notify.py]: Also, I don't see a `self` parameter in the method.\
Is this a static method?\
If so please add `@staticmethod` decorator, else add a `self` parameter in the function to make it correct.

Comment (BajajGirik) [reply src/notfication/notify.py]:
```python
def sendEmail(self, to:list[str], subject: str, message: str):
```

Something on these lines...

File: src/notfication/notify.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import os
+import ssl
+# from dotenv import load_dotenv
+# load_dotenv()
+
+class Notify:
+    # __serviceAccEmail = os.getenv('SERVICE_EMAIL')
+    # __serviceAccPassword = os.getenv('SERVICE_PASSWORD')
+
+    _subject = " Something went wrong !!"
```
Comment (BajajGirik) [inline src/notfication/notify.py]: Let's the caller handle sending the subject.\
We might want to also add the service name that crashed in the subject.\
Example: "Discord service crashed!"

File: src/notfication/notify.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+    def email(errorMessage:str,to:list[str]):
+
+        for dest in to:
+            em = EmailMessage()
```
Comment (BajajGirik) [inline src/notfication/notify.py]: Please rename this variable.

File: src/notfication/notify.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+            em['From']=Notify.__serviceAccEmail
+            em['Subject']= Notify._subject
+            em.set_content(errorMessage)
+            em['To']= dest
+            context = ssl.create_default_context()
+            with smtplib.SMTP_SSL('smtp.gmail.com', 465 , context=context) as s:    
+               
+                s.login(Notify.__serviceAccEmail, Notify.__serviceAccPassword)
```
Comment (BajajGirik) [inline src/notfication/notify.py]: Why are we performing the login multiple times in the for loop?\
The login event will be triggering before sending the email to each individual which is unnecessary.\
Can we move this to outside the for loop as we just need to login once?

**Review by BajajGirik — CHANGES_REQUESTED** [review]
Please resolve all the comments and update requirements.txt file as well

File: src/notfication/notify.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
+from email.message import EmailMessage
+import smtplib
+import os
+import ssl
+# from dotenv import load_dotenv
+# load_dotenv()
+
+class Notify:
```
Comment (BajajGirik) [inline src/notfication/notify.py]: Let's rename this something like `EmailNotification` and please change the file name to `email.py` as well.
As to why this change is needed, you can have a look at this [issue](https://github.com/BajajGirik/deal-finder/issues/10).

File: .env.example (hunk 1 above)
Comment (PhoenixOutlaw) [reply .env.example]: done

File: src/notfication/notify.py (hunk 4 above)
Comment (PhoenixOutlaw) [reply src/notfication/notify.py]: ok

File: src/notfication/notify.py (hunk 5 above)
Comment (PhoenixOutlaw) [reply src/notfication/notify.py]: ok

File: src/notfication/notify.py (hunk 6 above)
Comment (PhoenixOutlaw) [reply src/notfication/notify.py]: done

File: src/notfication/notify.py (hunk 7 above)
Comment (PhoenixOutlaw) [reply src/notfication/notify.py]: done

File: src/notfication/notify.py (hunk 8 above)
Comment (PhoenixOutlaw) [reply src/notfication/notify.py]: changing the name to email.py will lead to some dependency conflicts

**Comment by BajajGirik** [comment]
> Please resolve all the comments and update requirements.txt file as well

My bad. I don't see any new libraries added (smtp and email are already inbuilt in python)

File: src/notfication/notify.py (hunk 8 above)
Comment (BajajGirik) [reply src/notfication/notify.py]: Strange. I fixed and tested it out now.

**Review by BajajGirik — APPROVED** [review]
Tested it out and seems functional.\
Good Job completing the feature!!
